### ISSUE TITLE
when batch_count > 1 in `hipfftMakePlanMany`, it throws HIP_PARSE_ERROR in ROCm 5.1

### ISSUE THREAD

**Issue by yhmtsai** [ROCmSoftwarePlatform/rocFFT#353]
### What is the expected behavior
- it should not throw the error as other version (5.0.2, 4.5)

### What actually happens
- In ROCm 5.1, it will throws error. it does not get any plan created after several `ROC_FFT_CHECK_PLAN_CREATE`.

### How to reproduce
- create a fft.hip.cpp with the following content
- `hipcc fft.hip.cpp -lrocfft -lhipfft -L/opt/rocm/hipfft/lib -L/opt/rocm/lib -I/opt/rocm/hipfft/include`
- `./a.out`
```
#include <complex>
#include <iostream>
#include <vector>

#include <hip/hip_runtime.h>
#include <hipfft.h>

// hipcc fft.hip.cpp -lrocfft -lhipfft -L/opt/rocm/hipfft/lib -L/opt/rocm/lib -I/opt/rocm/hipfft/include
int main()
{
    std::cout << "hipfft 2D double-precision complex-to-complex transform\n";

    const int Nx        = 16 * 32;
    const int Ny        = 64;
    int       direction = HIPFFT_FORWARD; // forward=-1, backward=1

    std::vector<std::complex<double>> cdata(Nx * Ny);
    std::vector<int> fft_size{Nx, Ny};
    size_t complex_bytes = sizeof(decltype(cdata)::value_type) * cdata.size();

    // Create HIP device object and copy data to device:
    // hipfftComplex for single-precision
    hipError_t           hip_rt;
    hipfftDoubleComplex* x;
    hip_rt = hipMalloc(&x, complex_bytes);
    if(hip_rt != hipSuccess)
        throw std::runtime_error("hipMalloc failed");

    // Inititalize the data
    for(size_t i = 0; i < Nx * Ny; i++)
    {
        cdata[i] = i;
    }
    hip_rt = hipMemcpy(x, cdata.data(), complex_bytes, hipMemcpyHostToDevice);
    if(hip_rt != hipSuccess)
        throw std::runtime_error("hipMemcpy failed");

    // Create plan
    hipfftHandle plan      = NULL;
    hipfftResult hipfft_rt = hipfftCreate(&plan);
    if(hipfft_rt != HIPFFT_SUCCESS)
        throw std::runtime_error("failed to create plan");
    hipfft_rt = hipfftSetAutoAllocation(plan, false);
    if(hipfft_rt != HIPFFT_SUCCESS)
        throw std::runtime_error("failed to off auto allocation");
    std::size_t work_size = 0;
    hipfft_rt = hipfftMakePlanMany(
            plan, 2, fft_size.data(), fft_size.data(),
            3, 1, fft_size.data(),
            3, 1,
            HIPFFT_Z2Z,
            3, &work_size);
    if(hipfft_rt != HIPFFT_SUCCESS)
        throw std::runtime_error("hipfftMakePlanMany failed");

    hipfftDestroy(plan);
    hipFree(x);

    return 0;
}
```

### Environment
| Hardware | description |
|-----|-----|
| GPU | RadeonVII |
| CPU | AMD Ryzen Threadripper 1920X 12-Core Processor |

| Software | version |
|-----|-----|
| driver |  4.18.0-365.el8.x86_64 | 
| HSA runtime | 1.1 |
| Library | 5.1 or 4.5|

driver version is from `rocm-smi --showdriverversion` in the container
HSA runtime is from `rocminfo` 
I do not know how to get ROCR or ROCK version. If someone can give the commands, it will be helpful and I will check the version

**Comment by yhmtsai** [comment]
Thanks for fixing it.
I try it on rocm5.1 and rocm5.1.1, but it still throws the error.
Is this only included after future releases like 5.1.2?

**Comment by evetsso** [comment]
Currently the fix is only in the master and develop branches.  I'm going through the process to also get this fixed in 5.2.

The fix is not in any 5.1.x.

**Comment by yhmtsai** [comment]
I see. Thanks for explanation.

**Comment by evetsso** [comment]
I can confirm that this fix will land in 5.2.
